### PR TITLE
PS-5735: Handle linking of %{_datadir}/mysql directory properly

### PR DESCRIPTION
**Pull request by Sudokamikaze**
By default, CentOS 7 comes with mariadb-libs preinstalled as dependency for postfix. Re-worked scheme of linking /usr/share/percona-server to /usr/share/mysql as `if` check will always fail by default because of mariadb-libs existance and because order of processing package operations in RPM.

### CODE REVIEW

**Comment by Sudokamikaze** [comment]
I think we should include another if check if it's a symlink on upgrade procedure, as %pretrans will re-create it all the time. Like `if [ -L ]` in %pretrans

What do you think? @EvgeniyPatlan @vorsel

File: build-ps/percona-server.spec [hunk 1]
```diff
@@ -550,6 +550,18 @@ rm -rf %{buildroot}%{_bindir}/mysql_embedded
...
 %endif
 
+%pretrans -n Percona-Server-server%{product_suffix}
+if [ -d %{_datadir}/mysql ]; then
+  MYCNF_PACKAGE=$(rpm -qi `rpm -qf /usr/share/mysql` | grep -m 1 Name | awk '{print $3}')
+fi
+
+if [ "$MYCNF_PACKAGE" == "mariadb-libs" -o "$MYCNF_PACKAGE" == "mysql-libs" ]; then
+  MODIFIED=$(rpm -Va "$MYCNF_PACKAGE" | grep '/usr/share/mysql' | awk '{print $1}' | grep -c 5)
+  if [ "$MODIFIED" == 1 ]; then
```
Comment (EvgeniyPatlan) [inline build-ps/percona-server.spec:560]: it is better to add check that '/usr/share/mysql' is not link

**Review by EvgeniyPatlan — CHANGES_REQUESTED** [review]

File: build-ps/percona-server.spec (hunk 1 above)
Comment (Sudokamikaze) [reply build-ps/percona-server.spec:560]: Gotcha, will do

File: build-ps/percona-server.spec [hunk 2]
```diff
@@ -680,6 +686,11 @@ fi
   fi
 %endif
 
+%posttrans -n Percona-Server-server%{product_suffix}
+if [ ! -d %{_datadir}/mysql ]; then
```
Comment (EvgeniyPatlan) [inline build-ps/percona-server.spec]: and if it is link?

Comment (Sudokamikaze) [reply build-ps/percona-server.spec]: Gotcha, sorry for missing this point. Fixed

**Review by vorsel — APPROVED** [review]

**Review by EvgeniyPatlan — APPROVED** [review]

**Comment by Sudokamikaze** [comment]
Tested with new changes, works as expected and moves directory to old in %posttrans section

**Review by vorsel — APPROVED** [review]

**Review by EvgeniyPatlan — APPROVED** [review]

**Pull request merged**